src/auth/google/google-oauth.service.ts

```diff
@@ -56,8 +61,45 @@ export class GoogleOAuthService {
     }
   }
 
-  async login(token: string): Promise<GoogleOAuthLoginResponseDto> {
-    const userInfo = await this.authenticate(token);
+  async mainAuthenticate(
+    token: string,
+  ): Promise<{ user: User; picture: string }> {
+    try {
+      const tokenVerified = await this.googleOAuthClient.verifyIdToken({
+        idToken: token,
+        audience: this.googleClientID,
+      });
+      const { email, name, picture } = tokenVerified.getPayload();
+      let user = await this.userService.findOne({
+        where: { email: email },
+      });
+
+      // Create new user when user not exist
+      if (!user) {
+        user = await this.userService.create({
```

Comment:
remove duplicate function and consider suggest
```suggestion
        let role = USER_ROLE.USER;
        if (adminInitEmail === googleEmail) role = USER_ROLE.ADMIN;
        user = await this.userService.create({
```